[BUG] Reproduction code does not include values, passed manually

**Describe the bug**
If I pass some custom values manually to the `case.call` then they are not displayed in the failure reproduction code.

**To Reproduce**

```python
case.call(headers={"Authorization": "Bearer token"})
```

And the output:

```
E           Run this Python code to reproduce this response: 
E           
E               requests.patch('http://127.0.0.1:8081/api/users/2198', headers={'User-Agent': 'schemathesis/2.5.1'}, params={'common': 0}, json={'username': 'foo'})
```

No auth header (or any other header)

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -12,6 +12,10 @@ Changelog
 
 - The ``response_schema_conformance`` and ``content_type_conformance`` checks fail unconditionally if the input response has no ``Content-Type`` header. `#816`_
 
+**Fixed**
+
+- Failure reproduction code missing values that were explicitly passed to ``call_*`` methods during testing. `#814`_
+
 **Deprecated**
 
 - Using ``stateful=Stateful.links`` in schema loaders and ``parametrize``. Use ``schema.as_state_machine().TestCase`` instead.
@@ -1416,6 +1420,7 @@ Deprecated
 .. _0.2.0: https://github.com/schemathesis/schemathesis/compare/v0.1.0...v0.2.0
 
 .. _#816: https://github.com/schemathesis/schemathesis/issues/816
+.. _#814: https://github.com/schemathesis/schemathesis/issues/814
 .. _#795: https://github.com/schemathesis/schemathesis/issues/795
 .. _#793: https://github.com/schemathesis/schemathesis/issues/793
 .. _#788: https://github.com/schemathesis/schemathesis/issues/788

--- a/src/schemathesis/models.py
+++ b/src/schemathesis/models.py
@@ -119,10 +119,20 @@ class Case:  # pylint: disable=too-many-public-methods
             if (key == "Body" and value is not None) or value not in (None, {})
         ]
 
-    def get_code_to_reproduce(self, headers: Optional[Dict[str, Any]] = None) -> str:
+    def get_code_to_reproduce(
+        self, headers: Optional[Dict[str, Any]] = None, request: Optional[requests.PreparedRequest] = None
+    ) -> str:
         """Construct a Python code to reproduce this case with `requests`."""
-        base_url = self.get_full_base_url()
-        kwargs = self.as_requests_kwargs(base_url)
+        if request is not None:
+            kwargs: Dict[str, Any] = {
+                "method": request.method,
+                "url": request.url,
+                "headers": request.headers,
+                "body": request.body,
+            }
+        else:
+            base_url = self.get_full_base_url()
+            kwargs = self.as_requests_kwargs(base_url)
         if headers:
             final_headers = kwargs["headers"] or {}
             final_headers.update(headers)
@@ -254,6 +264,8 @@ class Case:  # pylint: disable=too-many-public-methods
         client = werkzeug.Client(application, WSGIResponse)
         with cookie_handler(client, self.cookies):
             response = client.open(**data, **kwargs)
+        requests_kwargs = self.as_requests_kwargs(base_url=self.get_full_base_url(), headers=headers)
+        response.request = requests.Request(**requests_kwargs).prepare()
         if self.feedback:
             self.store_response(response)
         return response
@@ -292,9 +304,10 @@ class Case:  # pylint: disable=too-many-public-methods
         if errors:
             exception_cls = get_grouped_exception(self.endpoint.verbose_name, *errors)
             formatted_errors = "\n\n".join(f"{idx}. {error.args[0]}" for idx, error in enumerate(errors, 1))
+            code = self.get_code_to_reproduce(request=response.request)
             raise exception_cls(
                 f"\n\n{formatted_errors}\n\n----------\n\n"
-                f"Run this Python code to reproduce this response: \n\n    {self.get_code_to_reproduce()}"
+                f"Run this Python code to reproduce this response: \n\n    {code}"
             )
 
     def call_and_validate(

--- a/src/schemathesis/utils.py
+++ b/src/schemathesis/utils.py
@@ -170,7 +170,8 @@ StringDatesYAMLLoader = make_loader("tag:yaml.org,2002:timestamp")
 
 
 class WSGIResponse(BaseResponse, JSONMixin):  # pylint: disable=too-many-ancestors
-    pass
+    # We store "requests" request to build a reproduction code
+    request: requests.PreparedRequest
 
 
 def get_requests_auth(auth: Optional[RawAuth], auth_type: Optional[str]) -> Optional[Union[HTTPDigestAuth, RawAuth]]:

--- a/test/specs/openapi/test_stateful.py
+++ b/test/specs/openapi/test_stateful.py
@@ -70,7 +70,7 @@ TestStateful.settings = settings(
     # ["POST", "GET", "PATCH", "GET", "PATCH"]
     result.assert_outcomes(failed=1)
     # And there should be Python code to reproduce the error in the PATCH call
-    result.stdout.re_match_lines([rf"E +requests\.patch\('{openapi3_base_url}/users/\d+'.+"])
+    result.stdout.re_match_lines([rf"E +requests\.patch\('{openapi3_base_url}/users/\d+.+"])
     # And the reproducing example should work
     first = result.outlines.index("Falsifying example:") + 1
     last = result.outlines.index("state.teardown()") + 1

--- a/test/test_pytest.py
+++ b/test/test_pytest.py
@@ -213,6 +213,6 @@ def test(case):
             r".+2. Received a response with a status code, which is not defined in the schema: 500",
             r".+Declared status codes: 200",
             r".+Run this Python code to reproduce this response:",
-            rf".+requests.get\('{openapi3_base_url}/failure', headers={{'User-Agent': '{USER_AGENT}'}}\)",
+            rf".+requests.get\('{openapi3_base_url}/failure', headers={{'User-Agent': '{USER_AGENT}'",
         ]
     )

